Formula/m/mit-scheme.rb

```diff
@@ -55,11 +49,17 @@ def install
     # not the correct type." Only Haswell and above appear to be impacted.
     # Reported 23rd Apr 2016: https://savannah.gnu.org/bugs/index.php?47767
     # NOTE: `unless build.bottle?` avoids overriding --bottle-arch=[...].
+
     ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
```

Comment:
Is this still needed?

Reply:
It's not necessary, nor does it make any difference for modern Apple Silicon macOS hardware and software versions. 

However, for legacy ones that are technically unsupported by homebrew such as the Retina MacBook Pro and the like, it might be needed since this formula makes CPU / microcode specific changes during the installation and build process.

We could remove it in a new commit if necessary.

Reply:
Let's guard it with a check for the macOS version then

Reply:
```rb
if MacOS.version < :monterey && Hardware::CPU.intel?
  ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
end
```
The above should work but `brew audit --strict mit-scheme` flagged it as a statement that should be nested underneath the next if statement so the below implementation satisfies the same:
```rb
if OS.mac? && MacOS.version < :monterey && Hardware::CPU.intel?
  ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
  system "./configure", "--prefix=#{buildpath}/staging", "--without-x", \
    "CC=/usr/bin/gcc", "CXX=/usr/bin/g++"
elsif OS.mac?
  system "./configure", "--prefix=#{buildpath}/staging", "--without-x", \
    "CC=/usr/bin/gcc", "CXX=/usr/bin/g++"
  else
    system "./configure", "--prefix=#{buildpath}/staging", "--without-x"
end
```

Reply:
I've removed it and all the CI/CD builds, including `macOS 12-x86_64` have passed so it doesn't seem necessary anymore.